lib/trento.pm

```diff
@@ -0,0 +1,366 @@
+                my $kubectl_describe_cmd = $self->az_vm_ssh_cmd("kubectl describe pods/$pod_name > $describe_txt", $machine_ip);
+                script_run($kubectl_describe_cmd, 180);
+                upload_logs($describe_txt);
+
+                # get the log
+                my $kubectl_logs_cmd = $self->az_vm_ssh_cmd("kubectl logs $pod_name > $log_txt", $machine_ip);
+                script_run($kubectl_logs_cmd, 180);
+                upload_logs($log_txt);
+            }
+        }
+    }
+    script_run('ls -lai *.txt', 180);
+}
+
+=head3 podman_self_check
+
+Perform some generic checks related to the podman installation itself
+and the relevant parameters of the current machine environment.
+=cut
+sub podman_self_check {
+    record_info('PODMAN', 'Check podman');
+    assert_script_run('which podman');
+    assert_script_run('podman --version');
+    assert_script_run('podman info --debug');
+    assert_script_run('podman ps');
+    assert_script_run('podman images');
+    assert_script_run('df -h');
+}
+
+=head3 cypress_install_container
+
+Prepare whatever is needed to run cypress tests using container
+
+=over 1
+
+=item B<CYPRESS_VER> - String used as tag for the cypress/included image.
+
+=back
+=cut
+sub cypress_install_container {
+    my ($self, $cypress_ver) = @_;
+    podman_self_check();
+
+    # list all the available cypress images
+    my $cypress_image = 'docker.io/cypress/included';
+    assert_script_run('podman search --list-tags ' . $cypress_image);
+
+    # Pull in advance the cypress container
+    my $podman_pull_cmd = 'time podman ' .
+      '--log-level trace ' .
+      'pull ' .
+      '--quiet ' .
+      $cypress_image . ':' . $cypress_ver .
+      ' | tee ' . PODMAN_PULL_LOG;
+    assert_script_run($podman_pull_cmd, 1800);
+    assert_script_run('df -h');
+    assert_script_run('podman images');
+}
+
+=head3 cypress_log_upload
+
+Upload to openQA the relevant logs
+
+=over 1
+
+=item B<LOG_FILTER> - List of strings. List of file extensions (dot needed) 
+
+=back
+=cut
+sub cypress_log_upload {
+    my ($self, @log_filter) = @_;
+    my $find_cmd = 'find ' . CYPRESS_LOG_DIR . ' -type f \( -iname \*' . join(' -o -iname \*', @log_filter) . ' \)';
+
+    upload_logs("$_") for split(/\n/, script_output($find_cmd));
+}
+
+=head3 cypress_exec
+
+Execute a cypress command within the container 
+
+=over 5
+
+=item B<CYPRESS_TEST_DIR> - String of the path where the cypress Trento code is available.
+It is the I<test> folder within the path used by L<setup_jumphost>
+
+=item B<CMD> - String of cmd to be used as main argument for the cypress
+executable call. 
+
+=item B<TIMEOUT> - Integer used as timeout for the cypress command execution
+
+=item B<LOG_PREFIX> - String of the command to be executed remotely
+
+=item B<FAILOK> - Integer boolean value. 0:test marked as failure if the podman/cypress
+return not 0 exit code. 1:all not 0 podman/cypress exit code are ignored. SoftFail reported.
+
+=back
+=cut
```

Comment:
I didn't see this earlier, but always leave a newline between pod and code. Some pod renderers will see this as malformed.

Reply:
I didn't know about that, but I've always instinctively left empty lines surrounding all pod command as in https://github.com/os-autoinst/os-autoinst-distri-opensuse/blob/master/lib/hacluster.pm or https://github.com/os-autoinst/os-autoinst-distri-opensuse/blob/master/lib/sles4sap.pm (my main contributions to pod documentation in this repo).

@mpagot it seems it would be a good idea to apply Tina's suggestion even though most libraries within os-autoinst-distri-opensuse are not following it (examples: https://github.com/os-autoinst/os-autoinst-distri-opensuse/blob/master/lib/susedistribution.pm, https://github.com/os-autoinst/os-autoinst-distri-opensuse/blob/master/lib/opensusebasetest.pm or https://github.com/os-autoinst/os-autoinst-distri-opensuse/blob/master/lib/utils.pm).

Reply:
I play a little bit with it locally using `pod2html` exactly about this topic: "Where new line is needed"
For example new line is strictly needed after =head and =index

Ignoring for a second what is strictly needed for a proper rendering, and just focusing of the generic code readability: leaving one empty like between pod and code is maybe fine but somehow collide (personal opinion) with https://github.com/os-autoinst/os-autoinst-distri-opensuse/pull/15049#discussion_r895862427  (leaving 2 new lines between each method).